### PR TITLE
 Move code from RemoteSession to SessionRepository, take 2

### PR DESCRIPTION
**Pull request by hmusum**
Move tests and avoid a lot of low-level setup code

Does not use `synchronized` for `ensureApplicationLoaded()`, otherwise same as https://github.com/vespa-engine/vespa/pull/14190 (which led to worse performance, the theory is that it was because of an unnecesary `synchronized` qualifier on `ensureApplicationLoaded()`

### CODE REVIEW

File: configserver/src/main/java/com/yahoo/vespa/config/server/ApplicationRepository.java.orig [hunk 1]
```diff
@@ -0,0 +1,1100 @@
...
+import static java.nio.file.Files.readAttributes;
+
+/**
+ * The API for managing applications.
+ *
+ * @author bratseth
+ */
+// TODO: Move logic for dealing with applications here from the HTTP layer and make this the persistent component
+//       owning the rest of the state
+public class ApplicationRepository implements com.yahoo.config.provision.Deployer {
```
Comment (jonmv) [inline configserver/src/main/java/com/yahoo/vespa/config/server/ApplicationRepository.java.orig:111]: Does not compile 🤖

Comment (hmusum) [reply configserver/src/main/java/com/yahoo/vespa/config/server/ApplicationRepository.java.orig]: Urgh, thanks for fixing.

**Review by jonmv — APPROVED** [review]

File: configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java [hunk 2]
```diff
@@ -394,13 +402,84 @@ private void loadSessionIfActive(RemoteSession session) {
...
+        if (applicationSet.isPresent()) {
+            return applicationSet.get();
+        }
+
+        ApplicationSet newApplicationSet = loadApplication(session);
+        RemoteSession newSession = new RemoteSession(session.getTenantName(),
+                                    session.getSessionId(),
+                                    session.getSessionZooKeeperClient(),
+                                    Optional.of(newApplicationSet));
+        remoteSessionCache.putSession(newSession);
```
Comment (jonmv) [inline configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java:430]: Sure this shouldn't be synchronized, but more fine-grained than on the session repository shared by all sessions?

Comment (hmusum) [reply configserver/src/main/java/com/yahoo/vespa/config/server/session/SessionRepository.java]: Yes, I think you are right. Need to look into that.

**Comment by hmusum** [comment]
Please review only, will merge later

**Review by jonmv — APPROVED** [review]

**Pull request merged**